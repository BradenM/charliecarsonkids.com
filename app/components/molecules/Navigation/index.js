/**
 *
 * Navigation
 *
 * Molecule
 */

import PropTypes from 'prop-types';
import React from 'react';
import Nav from 'react-bootstrap/Nav';
import Navbar from 'react-bootstrap/Navbar';
import { LinkContainer } from 'react-router-bootstrap';
import styled from 'styled-components';

const StyledNavbar = styled(Navbar)`
  .navbar {
    &-nav {
      letter-spacing: 0.15em;
      min-width: 75%;
    }
  }
`;

const NavLink = ({ text, ...props }) => (
  <LinkContainer {...props}>
    <Nav.Link>{text}</Nav.Link>
  </LinkContainer>
);

NavLink.propTypes = {
  text: PropTypes.string.isRequired,
};

const Navigation = () => (
  <StyledNavbar collapseOnSelect expand="lg" variant="light">
    <Navbar.Toggle aria-controls="responsive-navbar-nav" />
    <Navbar.Collapse id="responsive-navbar-nav">
      <Nav className="m-auto justify-content-around">
        <NavLink to="new" text="new" />
        <NavLink to="boys" text="boys" />
        <NavLink to="girls" text="girls" />
        <NavLink to="baby" text="baby" />
        <NavLink to="collections" text="collections" />
        <NavLink to="about" text="about" />
      </Nav>
    </Navbar.Collapse>
  </StyledNavbar>
);

Navigation.propTypes = {};

export default Navigation;
