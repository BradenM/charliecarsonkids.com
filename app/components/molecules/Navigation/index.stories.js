/**
 *
 * Stories for Navigation
 *
 * Molecule
 */

import React from 'react';
import StoryRouter from 'storybook-react-router';
import styled from 'styled-components';
import Navigation from '.';

const NavBackground = styled.div`
  padding-top: 5em;
  padding-bottom: 5em;
  background-color: gray;
`;

export default {
  title: 'Design|Molecules/Navigation',
  component: Navigation,
  decorators: [
    StoryRouter(),
    story => <NavBackground>{story()}</NavBackground>,
  ],
};

export const defaultNavigation = () => <Navigation />;
