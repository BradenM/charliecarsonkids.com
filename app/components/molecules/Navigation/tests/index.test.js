/**
 *
 * Tests for Navigation
 *
 * Molecules
 */

import { render } from '@testing-library/react';
import { ConnectedRouter } from 'connected-react-router/immutable';
import { createMemoryHistory } from 'history';
import React from 'react';
import { Provider } from 'react-redux';
import configureStore from '../../../../configureStore';
import Navigation from '../index';

const renderComponent = (store, history, props = {}) =>
  render(
    <Provider store={store}>
      <ConnectedRouter history={history}>
        <Navigation {...props} />
      </ConnectedRouter>
    </Provider>,
  );

describe('<Navigation />', () => {
  let store;
  let history;

  beforeAll(() => {
    history = createMemoryHistory();
    store = configureStore({}, history);
  });

  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    renderComponent(store, history);
    expect(spy).not.toHaveBeenCalled();
  });

  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = renderComponent(store, history);
    expect(firstChild).toMatchSnapshot();
  });
});
