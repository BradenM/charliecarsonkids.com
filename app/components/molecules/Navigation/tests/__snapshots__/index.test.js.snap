// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navigation /> Should render and match the snapshot 1`] = `
<nav
  class="Navigation__StyledNavbar-sc-13st3hq-0 kdGuE navbar navbar-expand-lg navbar-light"
>
  <button
    aria-controls="responsive-navbar-nav"
    aria-label="Toggle navigation"
    class="navbar-toggler collapsed"
    type="button"
  >
    <span
      class="navbar-toggler-icon"
    />
  </button>
  <div
    class="navbar-collapse collapse"
    id="responsive-navbar-nav"
  >
    <div
      class="m-auto justify-content-around navbar-nav"
    >
      <a
        class="nav-link"
        data-rb-event-key="new"
        href="new"
      >
        new
      </a>
      <a
        class="nav-link"
        data-rb-event-key="boys"
        href="boys"
      >
        boys
      </a>
      <a
        class="nav-link"
        data-rb-event-key="girls"
        href="girls"
      >
        girls
      </a>
      <a
        class="nav-link"
        data-rb-event-key="baby"
        href="baby"
      >
        baby
      </a>
      <a
        class="nav-link"
        data-rb-event-key="collections"
        href="collections"
      >
        collections
      </a>
      <a
        class="nav-link"
        data-rb-event-key="about"
        href="about"
      >
        about
      </a>
    </div>
  </div>
</nav>
`;
